### PR TITLE
Feature/party management 2

### PR DESCRIPTION
**Pull request by judgeaxl**
I think this is in a stage where it could be merged for functionality, while there certainly are details that should be sorted out, expiration for one, but I just want some more eyes on it.

### CODE REVIEW

File: driftbase/parties.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    return g.redis.conn.hgetall(scoped_party_invite_key)
+
+
+def decline_party_invite(invite_id, declining_player_id):
+    scoped_player_party_key = make_player_party_key(declining_player_id)
+    scoped_party_invite_key = make_party_invite_key(invite_id)
+
+    with g.redis.conn.pipeline() as pipe:
+        try:
+            pipe.watch(scoped_player_party_key)
```
Comment (einarth) [inline driftbase/parties.py:193]: Not sure I'm following here; why does this key need to be watched?  Are you guarding against the case where the player joins the party while this method is running?

File: driftbase/api/parties.py [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+
+        member_query = g.db.query(CorePlayer.player_id, CorePlayer.player_name).filter(CorePlayer.player_id.in_(members))
+        response, response_header = make_party_response(party_id, member_query.all())
+        return response, http_client.OK, response_header
+
+    def delete(self, party_id):
+        members = get_party_members(party_id)
+        for member in members:
+            leave_party(party_id, member)
+        return {}, http_client.NO_CONTENT
```
Comment (einarth) [inline driftbase/api/parties.py:321]: Do we not want to delete the party after removing the last player from it?

Comment (judgeaxl) [reply driftbase/api/parties.py:321]: Yes that would make sense, fixing...

File: driftbase/parties.py (hunk 1 above)
Comment (judgeaxl) [reply driftbase/parties.py:193]: This is probably wrong. I'm also considering just replacing most of these watches with locks

Comment (judgeaxl) [reply driftbase/parties.py:193]: I'll remove that watch, can't think of anything that really need to be watched for this operation

**Pull request merged**